ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcRawRecordMerger.java

```diff
@@ -763,15 +763,20 @@ private KeyInterval discoverOriginalKeyBounds(Reader reader, int bucket,
   }
 
   /**
-   * Find the key range for the split (of the base).  These are used to filter delta files since
-   * both are sorted by key.
+   * Find the key range for the split (of the base) based on the 'hive.acid.key.index' metadata.
+   * These keys are used to filter delta files since both are sorted by key.
+   * If 'hive.acid.key.index' is missing from the ORC file, return null keys (which forces a full read).
    * @param reader the reader
    * @param options the options for reading with
-   * @throws IOException
    */
-  private KeyInterval discoverKeyBounds(Reader reader,
-                                 Reader.Options options) throws IOException {
-    RecordIdentifier[] keyIndex = OrcRecordUpdater.parseKeyIndex(reader);
+  private KeyInterval discoverKeyBounds(Reader reader, Reader.Options options) {
+    final RecordIdentifier[] keyIndex = OrcRecordUpdater.parseKeyIndex(reader);
+    if (keyIndex == null) {
+      LOG.warn("Missing '{}' metadata in ORC acid file, can't compute min/max keys",
```

Comment:
nit: 'Instead of ORC 'acid' file, suggest just using ORC file since the table has acid property, not the file.